src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/r/SignatureToolTipManager.java

```diff
@@ -81,9 +145,45 @@ public void initialize(UIPrefs uiPrefs,
       server_ = server;
    }
    
+   private void attachPreviewHandler()
+   {
+      detachPreviewHandler();
+      preview_ = Event.addNativePreviewHandler(new NativePreviewHandler()
+      {
+         @Override
+         public void onPreviewNativeEvent(NativePreviewEvent preview)
+         {
+            if (preview.getTypeInt() == Event.ONMOUSEMOVE)
+            {
+               NativeEvent event = preview.getNativeEvent();
+               coordinates_ = new MouseCoordinates(
+                     event.getClientX(),
+                     event.getClientY());
+            }
+            else if (preview.getTypeInt() == Event.ONKEYDOWN)
+            {
+               coordinates_ = null;
+            }
+            
+            timer_.schedule(DELAY_MS);
```

Comment:
I would worry about it when it's married to a native event handler which
could be firing dozens or hundreds of times a second (depending on what
mouse events are getting sent)

On Tue, Mar 1, 2016 at 6:16 PM, Kevin Ushey <notifications@github.com>
wrote:

> In
> src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/r/SignatureToolTipManager.java
> <https://github.com/rstudio/rstudio/pull/614#discussion_r54653184>:
>
> > +         @Override
> > +         public void onPreviewNativeEvent(NativePreviewEvent preview)
> > +         {
> > +            if (preview.getTypeInt() == Event.ONMOUSEMOVE)
> > +            {
> > +               NativeEvent event = preview.getNativeEvent();
> > +               coordinates_ = new MouseCoordinates(
> > +                     event.getClientX(),
> > +                     event.getClientY());
> > +            }
> > +            else if (preview.getTypeInt() == Event.ONKEYDOWN)
> > +            {
> > +               coordinates_ = null;
> > +            }
> > +
> > +            timer_.schedule(DELAY_MS);
>
> This makes sense, but do we really need to worry about the cost of
> scheduling a timer? (I imagine that's a fairly cheap operation).
>
> @jmcphers <https://github.com/jmcphers> is definitely right pointing out
> that we don't want to re-schedule on *all* events, though.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rstudio/rstudio/pull/614/files#r54653184>.
>